MediaRecorder.bs

```diff
@@ -286,11 +276,35 @@ interface MediaRecorder : EventTarget {
     <li>Set |recorder|'s {{state}} to {{recording}}, and run the following steps
     in parallel:
     <ol>
-      <li>Start recording all tracks in |tracks| using the {{MediaRecorder}}'s
-      current configuration, and gather the data into a {{Blob}} <var>blob</var>
-      and queue a task, using the DOM manipulation task source, to
-      <a>fire an event</a> named <a>start</a> at <var>recorder</var>, then return
-      <code>undefined</code>.</li>
+      <li>Let the User Agent specify the |recorder|'s configuration for
+      recording all the tracks in |tracks|, given its current constraints. If
+      track data is available, the User Agent MAY take the content into account
+      when making this decision as long as this does not violate how the
+      configuration has previously been constrained (see earlier steps).</li>
+      <div class="note">
+        By looking at the content of the tracks when recording start, the User
+        Agent could take frame information into account. For example, if the
+        MediaStreamTrack is a remote track sourced from an RTCPeerConnection,
+        the User Agent is allowed to pick the same codec that the source RTP
+        stream uses, which may allow the User Agent to avoid the cost of
+        re-encoding by recording the raw encoded frames.
```

Comment:
I think this needs rephrasing because the encoding of the rtp stream is not inherently part of "frame information".

Maybe "By looking at the sources of the tracks in |tracks| when recording starts, the User Agent could choose a configuration that avoids re-encoding track content. For example, if the MediaStreamTrack is a remote track sourced from an RTCPeerConnection, the User Agent could pick the same codec as is used for the MediaStreamTrack's RTP stream, should it be known."?

And perhaps mention that the RTP stream codec could change on the fly so the MediaRecorder must at least be able to encode it itself, to avoid an interrupted recording. Choosing a codec like that seems forbidden by the "For each track in |tracks|, if the User Agent cannot record the track using the current configuration, then throw a {{NotSupportedError}} {{DOMException}} and abort all steps." above, but it's probably good to re-iterate on that.

Reply:
I like your way of phrasing this better than mine and by saying "sources" we don't close any doors unnecessarily (e.g. only a single codec being negotiated).

I also added a sentence about codec changing on the fly. Our implementer would probably like to fire "onerror" if the codec changes on the fly, but I think a change like that would require a separate issue and discussions. For now it says "if the codec of the RTP stream changes while recording the User Agent have to be prepared to re-encode it to avoid interruptions" so that UAs will produce a recording for the entire stream whether or not re-enocoding was ever avoided.